reinsurancecontract.py

```diff
@@ -63,7 +63,19 @@ def mature(self, time):
         else: #TODO: ? Instead: if self.insurancetype == "proportional":
             self.contract.unreinsure()
 
-        if np.random.uniform(0,1,1) < 0.95:
+        if np.random.uniform(0,1,1) < 1:
             reinrisk = self.property_holder.create_reinrisk(time,self.category)
-            if reinrisk is not None and hasattr(self.insurer, 'reinrisks_kept'):
-                self.insurer.reinrisks_kept.append(reinrisk)
\ No newline at end of file
+
+            if reinrisk is not None:
+                new_nonproportional_risks = [reinrisk]
+                [reinrisks_per_categ, number_reinrisks_categ] = self.insurer.risks_reinrisks_organizer(new_nonproportional_risks)  #Here the new reinrisks are organized by category.
+
+                for repetition in range(self.insurer.recursion_limit):     # TODO: find an efficient way to stop the recursion if there are no more risks to accept or if it is not accepting any more over several iterations.
+                    former_reinrisks_per_categ = copy.copy(reinrisks_per_categ)
+                    [reinrisks_per_categ, not_accepted_reinrisk] = self.insurer.process_newrisks_reinsurer(reinrisks_per_categ, number_reinrisks_categ, time)  #Here we process all the new reinrisks in order to keep the portfolio as balanced as possible.
+                    if former_reinrisks_per_categ == reinrisks_per_categ:   #Stop condition implemented. Might solve the previous TODO.
```

Comment:
I am uncertain whether it is efficient to compare lists of objects. Is it?

Reply:
We are doing that in metainsuranceorg.py in a much more intensive way. Here is only for one contract.